net-dns/knot/knot-2.9.1.ebuild

```diff
@@ -0,0 +1,103 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit systemd
+
+DESCRIPTION="High-performance authoritative-only DNS server"
+HOMEPAGE="https://www.knot-dns.cz/"
+SRC_URI="https://secure.nic.cz/files/knot-dns/${P/_/-}.tar.xz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+KNOT_MODULES=(
+	"+cookies"
+	"+dnsproxy"
+	"dnstap"
+	"geoip"
+	"+noudp"
+	"+onlinesign"
+	"+queryacl"
+	"+rrl"
+	"+stats"
+	"+synthrecord"
+	"+whoami"
+)
+IUSE="doc caps +fastparser idn +libidn2 systemd +utils ${KNOT_MODULES[@]}"
+
+RDEPEND="
+	acct-group/knot
+	acct-user/knot
+	dev-db/lmdb
+	dev-libs/libedit
+	dev-libs/userspace-rcu:=
+	dev-python/lmdb
+	net-libs/gnutls:=
+	caps? ( sys-libs/libcap-ng )
+	dnstap? (
+		dev-libs/fstrm
+		dev-libs/protobuf-c:=
+	)
+	geoip? ( dev-libs/libmaxminddb:= )
+	idn? (
+		!libidn2? ( net-dns/libidn:0= !net-dns/libidn2 )
+		libidn2? ( net-dns/libidn2:= )
+	)
+	systemd? ( sys-apps/systemd:= )
+"
+DEPEND="${RDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	doc? ( dev-python/sphinx )
+"
+
+S="${WORKDIR}/${P/_/-}"
+
+src_configure() {
+	local u
+	local my_conf=(
+		--with-storage="${EPREFIX}/var/lib/${PN}"
+		--with-rundir="${EPREFIX}/var/run/${PN}"
+		$(use_enable fastparser)
+		$(use_enable dnstap)
+		$(use_enable doc documentation)
+		$(use_enable utils utilities)
+		--enable-systemd=$(usex systemd)
+		$(use_with idn libidn)
+	)
+
+	for u in "${KNOT_MODULES[@]#+}"; do
+		my_conf+=("$(use_with ${u} module-${u})")
+	done
+
+	econf "${my_conf[@]}"
+}
+
+src_compile() {
+	default
+
+	if use doc; then
+		emake -C doc html
+		HTML_DOCS=( doc/_build/html/{*.html,*.js,_sources,_static} )
```

Comment:
You shouldn't modify global variables inside a phase like this. Would it work, if you moved this to `src_install` and called `local HTML_DOCS=( doc/_build/html/{*.html,*.js,_sources,_static} )` before `default` there?

Reply:
Seems to work:
```
./usr/share/doc/knot-2.9.1/
./usr/share/doc/knot-2.9.1/NEWS.bz2
./usr/share/doc/knot-2.9.1/README.bz2
./usr/share/doc/knot-2.9.1/html/
./usr/share/doc/knot-2.9.1/html/_static/
./usr/share/doc/knot-2.9.1/html/_static/pygments.css
./usr/share/doc/knot-2.9.1/html/_static/underscore.js
./usr/share/doc/knot-2.9.1/html/_static/basic.css
./usr/share/doc/knot-2.9.1/html/_static/comment.png
./usr/share/doc/knot-2.9.1/html/_static/comment-close.png
./usr/share/doc/knot-2.9.1/html/_static/documentation_options.js
./usr/share/doc/knot-2.9.1/html/_static/websupport.js
./usr/share/doc/knot-2.9.1/html/_static/jquery.js
./usr/share/doc/knot-2.9.1/html/_static/file.png
./usr/share/doc/knot-2.9.1/html/_static/ajax-loader.gif
./usr/share/doc/knot-2.9.1/html/_static/comment-bright.png
./usr/share/doc/knot-2.9.1/html/_static/doctools.js
./usr/share/doc/knot-2.9.1/html/_static/searchtools.js
./usr/share/doc/knot-2.9.1/html/_static/minus.png
./usr/share/doc/knot-2.9.1/html/_static/down.png
./usr/share/doc/knot-2.9.1/html/_static/up.png
./usr/share/doc/knot-2.9.1/html/_static/plus.png
./usr/share/doc/knot-2.9.1/html/_static/up-pressed.png
./usr/share/doc/knot-2.9.1/html/_static/underscore-1.3.1.js
./usr/share/doc/knot-2.9.1/html/_static/jquery-3.2.1.js
./usr/share/doc/knot-2.9.1/html/_static/down-pressed.png
./usr/share/doc/knot-2.9.1/html/_static/nature.css
./usr/share/doc/knot-2.9.1/html/_static/admon_note_48.png
./usr/share/doc/knot-2.9.1/html/_static/admon_warning_48.png
./usr/share/doc/knot-2.9.1/html/_static/admons.css
./usr/share/doc/knot-2.9.1/html/_static/admon_tip_48.png
./usr/share/doc/knot-2.9.1/html/_static/main.css
./usr/share/doc/knot-2.9.1/html/_static/admon_important_48.png
./usr/share/doc/knot-2.9.1/html/_static/admon_caution_48.png
./usr/share/doc/knot-2.9.1/html/_static/logo.svg
./usr/share/doc/knot-2.9.1/html/_sources/
./usr/share/doc/knot-2.9.1/html/_sources/appendices.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/configuration.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/index.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/installation.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/introduction.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_kdig.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_keymgr.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_khost.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_kjournalprint.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_knotc.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_knotd.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_knsec3hash.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_knsupdate.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/man_kzonecheck.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/migration.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/modules.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/operation.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/reference.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/requirements.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/troubleshooting.rst.txt
./usr/share/doc/knot-2.9.1/html/_sources/utilities.rst.txt
./usr/share/doc/knot-2.9.1/html/searchindex.js
./usr/share/doc/knot-2.9.1/html/utilities.html
./usr/share/doc/knot-2.9.1/html/troubleshooting.html
./usr/share/doc/knot-2.9.1/html/search.html
./usr/share/doc/knot-2.9.1/html/requirements.html
./usr/share/doc/knot-2.9.1/html/reference.html
./usr/share/doc/knot-2.9.1/html/operation.html
./usr/share/doc/knot-2.9.1/html/modules.html
./usr/share/doc/knot-2.9.1/html/migration.html
./usr/share/doc/knot-2.9.1/html/man_kzonecheck.html
./usr/share/doc/knot-2.9.1/html/man_knsupdate.html
./usr/share/doc/knot-2.9.1/html/man_knsec3hash.html
./usr/share/doc/knot-2.9.1/html/man_knotd.html
./usr/share/doc/knot-2.9.1/html/man_knotc.html
./usr/share/doc/knot-2.9.1/html/man_kjournalprint.html
./usr/share/doc/knot-2.9.1/html/man_khost.html
./usr/share/doc/knot-2.9.1/html/man_keymgr.html
./usr/share/doc/knot-2.9.1/html/man_kdig.html
./usr/share/doc/knot-2.9.1/html/introduction.html
./usr/share/doc/knot-2.9.1/html/installation.html
./usr/share/doc/knot-2.9.1/html/index.html
```